app/src/main/java/com/hover/stax/transfers/NonTemplateVariableAdapter.kt

```diff
@@ -0,0 +1,82 @@
+package com.hover.stax.transfers
+
+import android.annotation.SuppressLint
+import android.content.Context
+import android.text.Editable
+import android.text.TextWatcher
+import android.view.LayoutInflater
+import android.view.ViewGroup
+import androidx.recyclerview.widget.DiffUtil
+import androidx.recyclerview.widget.ListAdapter
+import androidx.recyclerview.widget.RecyclerView
+import com.hover.stax.R
+import com.hover.stax.databinding.NonTemplateVariableItemBinding
+import com.hover.stax.views.AbstractStatefulInput
+
+class NonTemplateVariableAdapter(private var variables: List<NonTemplateVariable>, private val editTextListener: NonTemplateVariableInputListener) :
+        ListAdapter<NonTemplateVariable, NonTemplateVariableAdapter.ViewHolder>(NonTemplateDiffCallback()) {
+
+    @SuppressLint("NotifyDataSetChanged")
+    fun updateStates(variables: List<NonTemplateVariable>) {
+        this.variables = variables
+        notifyDataSetChanged()
+    }
+
+    inner class ViewHolder(val binding: NonTemplateVariableItemBinding): RecyclerView.ViewHolder(binding.root) {
+        fun bindItems(nonTemplateVariable: NonTemplateVariable) {
+
+            val inputTextWatcher: TextWatcher = object : TextWatcher {
+                override fun beforeTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {}
+                override fun afterTextChanged(editable: Editable) {}
+                override fun onTextChanged(charSequence: CharSequence, i: Int, i1: Int, i2: Int) {
+                    nonTemplateVariable.value = charSequence.toString().replace(",".toRegex(), "")
+                    editTextListener.nonTemplateVariableInputUpdated (nonTemplateVariable)
+                }
+            }
+
+            binding.variableInput.addTextChangedListener(inputTextWatcher)
+            binding.variableInput.setHint(nonTemplateVariable.key)
+            binding.variableInput.tag = nonTemplateVariable.key
+            binding.variableInput.text = nonTemplateVariable.value
+
+            nonTemplateVariable.editTextState?.let {
```

Comment:
Instead of storing the state in an object just move the validation call to this class. Then it can set the state of each variable directly instead of reloading all the data after the validation is done.